dronegen/yum.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"path"
+)
+
+// This function calls the build-apt-repos tool which handles the APT portion of RFD 0058.
+func promoteYumPipeline() pipeline {
+	return getYumPipelineBuilder().buildPromoteOsPackagePipeline()
+}
+
+func migrateYumPipeline(triggerBranch string, migrationVersions []string) pipeline {
+	return getYumPipelineBuilder().buildMigrateOsPackagePipeline(triggerBranch, migrationVersions)
+}
+
+func getYumPipelineBuilder() *OsPackageToolPipelineBuilder {
+	optpb := NewOsPackageToolPipelineBuilder(
+		"drone-s3-yumrepo-pvc",
+		"centos:7", // This can probably be bumped but newer versions don't appear to have createrepo for some reason
```

Comment:
That's a bit scary. It looks like CentOS 8 is already EoL and they took the repo offline :/

If I understand things correctly, I don't think this pipeline actually builds any artifacts? If so, maybe we should consider `rockylinux:9` or something? Most distros still package `createrepo_c` / `createrepo-c`, so we could probably use Ubuntu if we wanted.

Reply:
Nope, the pipeline does not actually build any artifacts - only YUM repos. I'll bump this to `ubuntu:22.04` which provides `createrepo_c` via the `createrepo-c` package